src/predicates/propPathEq.js

```diff
@@ -0,0 +1,35 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Karthik Iyengar (karthikiyengar) */
+
+const curry = require('../core/curry')
+const isArray = require('../core/isArray')
+const isInteger = require('../core/isInteger')
+const isNil= require('../core/isNil')
+const isString = require('../core/isString')
+const equals = require('../core/equals')
+const Maybe = require('../core/Maybe')
```

Comment:
Do not think we need to go the `Maybe` route here. Using the types add a couple "cycles" for lifting and building the type. While this is just fine on the "user of crocks" end to capture the safety of the `Maybe` against whatever predicate they desire. I feel that as a lib, we should probably go the route of just "doing it by hand" without the `Maybe`. just iterate down.

This gets me thinking though, and this is out of scope for this PR, but maybe we should do for loops in these instances so we get an early exit on a long list and do that everywhere we path down. Do not worry about that for this, we can go back later and do all these places at once.

But what I am thinking for this PR is something like:

```js
path.reduce((acc, key) => {
  if (!(isString(key) || isInteger(key))) {
    throw new TypeError(
      'propPathEq: Array of strings or integers required for first argument'
    )
  }

  if(isNil(acc)) {
    return acc
  }

  return acc[key]
})
```

(or something like that, I did not run the code, but the gist is that if we are a value that can error from us trying to reference properties like `null` and `undefined`, we just keep whatever that value is and pass it down. `NaN` is included in `isNil`, but that is going to be a huge edge case if someone wants to access `NaN`s properties.

Thoughts?

Reply:
With regards to introducing an early-exit loop, I am all down for that, and I'll be updating my PR with the modified logic.

With regards to `if (isNil(acc)) { return acc}`, I think it would be semantically better to return an `undefined` instead. This avoids the edge case with `NaN`.